src/App.tsx

```diff
@@ -14,210 +45,224 @@ export const App: React.FC = () => {
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const handleSubmitForm = useCallback((
+    event?: React.FormEvent<HTMLFormElement>,
+  ) => {
+    if (!event) {
+      return;
+    }
+
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage('Title can not be empty');
+      setTitle('');
+
+      return;
+    }
+
+    const addNewTodo = async () => {
+      setIsAdding(true);
+
+      if (user) {
+        try {
+          setTempTodo({
+            id: 0,
+            userId: user?.id,
+            title: title.trim(),
+            completed: false,
+          });
+
+          const addedTodo = await addTodo({
+            userId: user?.id,
+            title: title.trim(),
+            completed: false,
+          });
+
+          setTitle('');
+
+          setTodos(currentTodos => [
+            ...currentTodos,
+            addedTodo,
+          ]);
+        } catch (error) {
+          setErrorMessage('Unable to add a todo');
+        } finally {
+          setIsAdding(false);
+          setTempTodo(null);
+        }
+      }
+    };
+
+    addNewTodo();
+  }, [title]);
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      setIsTodoDeleting(true);
+      setSelectedTodosId(todosIds => [
+        ...todosIds,
+        todoId,
+      ]);
+
+      setErrorMessage('');
+
+      await deleteTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+
+      setSelectedTodosId(todosIds => todosIds.filter(id => id !== todoId));
+    } catch (error) {
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setIsTodoDeleting(false);
+    }
+  }, []);
+
+  const removeCompletedTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        removeTodo(todo.id);
+      }
+    });
+  }, [todos]);
+
+  const changeTodo = useCallback(async (
+    todoId: number,
+    dataToUpdate: Partial<Todo>,
+  ) => {
+    try {
+      setIsTodoUpdating(true);
+      setErrorMessage('');
+      setSelectedTodosId(todosIds => [
+        ...todosIds,
+        todoId,
+      ]);
+
+      await updateTodo(
+        todoId,
+        dataToUpdate,
+      );
+
+      setTodos(currentTodos => currentTodos.map(todo => {
+        if (todo.id !== todoId) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          ...dataToUpdate,
+        };
+      }));
+
+      setSelectedTodosId(todosIds => todosIds.filter(id => id !== todoId));
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+    } finally {
+      setIsTodoUpdating(false);
+    }
+  }, []);
+
+  const isAllTodosCompleted = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const updateAllTodos = useCallback(() => {
+    todos.forEach(todo => {
+      const isTodoNeedToUpdate = !isAllTodosCompleted && !todo.completed;
+
+      if (isTodoNeedToUpdate || isAllTodosCompleted) {
```

Comment:
The function has name `updateAll` but I see some condition here, looks like you don't update all todos. Figure out how to clarify function name